### PR TITLE
Fix a bug whereby reusable CPUs and devices were not being honored

### PR DESCRIPTION
**Pull request by klueska**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change

/kind bug

> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
Previously, it was possible for reusable CPUs and reusable devices (i.e.
those previously consumed by init containers) to not be reused by
subsequent init containers or app containers if the TopologyManager was
enabled. This would happen because hint generation for the
TopologyManager was not considering the reusable devices when it made
its hint calculation.

As such, it would sometimes:
1) Generate a hint for a differnent NUMA node, causing the CPUs and
devices to be allocated from that node instead of the one where the
reusable devices live; or
2) End up thinking there were not enough CPUs or devices to allocate and
throw a TopologyAffinity admission error

This patch fixes this by ensuring that reusable CPUs and devices are
considered as part of TopologyHint generation. This frunctionality is
difficult to unit test since it spans multiple components, but an e2e
test will be added in a subsequent patch to test this functionality.

**Does this PR introduce a user-facing change?**:

```release-note
Fixed a bug whereby the allocation of reusable CPUs and devices was not being honored when the TopologyManager was enabled
```

### CODE REVIEW

**Comment by derekwaynecarr** [comment]
we should prepare a cherrypick for this as well.

/approve
/lgtm

**Review by nolancon — COMMENTED** [review]
Two small nits. LGTM otherwise

File: pkg/kubelet/cm/topologymanager/bitmask/bitmask.go [hunk 1]
```diff
@@ -120,6 +121,16 @@ func (s *bitMask) IsSet(bit int) bool {
 	return (*s & (1 << uint64(bit))) > 0
 }
 
+// AnysSet checks bit in mask to see if any provided bit is set to one
```
[the remark this answers is not in the document]
Comment (klueska) [reply pkg/kubelet/cm/topologymanager/bitmask/bitmask.go]: Fixed

File: pkg/kubelet/cm/topologymanager/bitmask/bitmask.go [hunk 2]
```diff
@@ -33,6 +33,7 @@ type BitMask interface {
 	IsEqual(mask BitMask) bool
 	IsEmpty() bool
 	IsSet(bit int) bool
+	AnySet(bitss []int) bool
```
[the remark this answers is not in the document]
Comment (klueska) [reply pkg/kubelet/cm/topologymanager/bitmask/bitmask.go]: Fixed

**Pull request merged**